Userland/Libraries/LibIMAP/Parser.cpp

```diff
@@ -716,12 +716,12 @@ FetchCommand::DataItem Parser::parse_fetch_data_item()
             consume("]");
         } else if (is_ascii_digit(section_type[0])) {
             data_item.section->type = FetchCommand::DataItem::SectionType::Parts;
-            data_item.section->parts = Vector<int>();
+            data_item.section->parts = Vector<unsigned>();
 
             while (!try_consume("]")) {
                 auto num = parse_number();
                 if (num != (unsigned)-1) {
```

Comment:
Looks like I messed up :P. This should be `try_parse_number()` which does actually return an Optional.

Reply:
Done :+1: